**Assignment 3**

 1.   **Name: Yaqin issa**
 2.   **ID: 1201576**




In [ ]:
import numpy as np

In [ ]:
from os import replace
np.random.seed(1201576)

population=np.random.normal(loc=1500,scale=700,size=10000000)


sample10=np.random.choice(population,size=10,replace=False)
sample100=np.random.choice(population,size=100,replace=False)
sample1000=np.random.choice(population,size=1000,replace=False)
sample10000=np.random.choice(population,size=10000,replace=False)

# print(sample10)
# print(sample100)
# print(sample1000)
# print(sample10000)




**sample error**




# What is the relation between the sample size and the sampling error?


A larger sample size (10000) is more likely to be representative of the
population compared to a smaller sample size (10) because it captures a wider range of values from the population.



In [ ]:

def mean(sample):
    sample_mean=np.mean(sample)

    return sample_mean;


print(f"population _mean ={mean(population)}")
print(f"10 _mean ={mean(sample10)}")
print(f"100 _mean ={mean(sample100)}")
print(f"1000 _mean ={mean(sample1000)}")
print(f"10000 _mean ={mean(sample10000)}")



population _mean =1500.1719282170518
10 _mean =1240.7176415571769
100 _mean =1553.4565055667358
1000 _mean =1505.985163673422
10000 _mean =1492.7358953749417


In [ ]:
def sampling_error(population_parameter, sample_statistic):

    return population_parameter - sample_statistic
# sample error
print(f"10 -SE ={sampling_error(mean(population),mean(sample10))}")
print(f"100 _SE ={sampling_error(mean(population),mean(sample100))}")
print(f"1000 _SE ={sampling_error(mean(population),mean(sample1000))}")
print(f"10000 _SE ={sampling_error(mean(population),mean(sample10000))}")

10 -SE =259.4542866598749
100 _SE =-53.284577349684014
1000 _SE =-5.813235456370194
10000 _SE =7.43603284211008


# Which sample is more representative of the population?

The sampling error represents the difference between the population parameter
( population mean) and the sample statistic ( sample mean). A negative sampling error means that the sample mean underestimates the population mean, while a positive sampling error means that the sample mean overestimates the population mean.

Based on the sampling errors :


The sample size of **10000** has the smallest sampling error (**7.43**), indicating that its sample mean is closest to the population mean. Therefore, the sample size of 10000 is **more representative of the population **compared to the other sample sizes.

In [ ]:
def analyze_sample(sample):

 df = len(sample) - 1
 randomness_test = np.random.chisquare(df, size=1)

 is_random = randomness_test > 0.05

 q1 = np.percentile(sample, 25)
 q2 = np.percentile(sample, 50)  # Median
 q3 = np.percentile(sample, 75)

 return is_random, q2, q1, q3


is_random, median, q1, q3 = analyze_sample(sample10000)
print("Sample 10000 analysis:")
print("Is random:", is_random)
print("Median:", median)
print("Q1:", q1)
print("Q3:", q3)



Sample 10000 analysis:
Is random: [ True]
Median: 1493.0618244687382
Q1: 1029.6871952482597
Q3: 1962.5061160259818



# Random distribution:

If is_random is greater than 0.05 (p-value > 0.05), the code assigns True to is_random. This indicates that the Chi-square test result suggests the sample might be random. A higher p-value means the observed distribution aligns more closely with what we would expect from a random distribution.


**b. What is the value that 50% of the data below it?**
Median: 1493.0618244687382


**d. What is the value that 25% of the data below it?**
Q1: 1029.6871952482597


**c. What is the value that 75% of the data below it?**
Q3: 1962.5061160259818

#What are the values calculated in b, c, and d are statistically called ?
50th percentile (median): This is the value that exactly half 50% of the data falls below and the other half falls above.

75th percentile (Q3 or upper quartile): This value indicates that 75% of the data falls below it, and the remaining 25% falls above.

25th percentile (Q1 or lower quartile): This value signifies that 25% of the data falls below it, and 75% falls above.